website/src/components/Survey/LabelInputGroup.tsx

```diff
@@ -0,0 +1,198 @@
+    ],
+    inverted: true,
+  },
+  sexual_content: {
+    zeroText: "Non Sexual",
+    zeroDescription: [],
+    oneText: "Sexual Content",
+    oneDescription: ["Contains sexual content"],
+    inverted: true,
+  },
+  toxicity: {
+    zeroText: "Non Toxic",
+    zeroDescription: [],
+    oneText: "Rude / Toxic",
+    oneDescription: ["Contains rude, abusive, profane or insulting content"],
+    inverted: true,
+  },
+  moral_judgement: {
+    zeroText: "Non-Judgemental",
+    zeroDescription: [],
+    oneText: "Judgemental",
+    oneDescription: ["Expresses moral judgement"],
+    inverted: true,
+  },
+  political_content: {
+    zeroText: "Apolitical",
+    zeroDescription: [],
+    oneText: "Political",
+    oneDescription: ["Expresses political views"],
+    inverted: true,
+  },
+  humor: {
+    zeroText: "Serious",
+    zeroDescription: [],
+    oneText: "Humorous / Sarcastic",
+    oneDescription: ["Contains humorous content including sarcasm"],
+    inverted: false,
+  },
+  hate_speech: {
+    zeroText: "Safe",
+    zeroDescription: [],
+    oneText: "Hateful",
+    oneDescription: [
+      "Content is abusive or threatening and expresses prejudice against a protected characteristic",
+      "Prejudice refers to preconceived views not based on reason. Protected characteristics include gender, ethnicity, religion, sexual orientation, and similar characteristics.",
+    ],
+    inverted: true,
+  },
+  threat: {
+    zeroText: "Safe",
+    zeroDescription: [],
+    oneText: "Contains Threat",
+    oneDescription: ["Contains a threat against a person or persons"],
+    inverted: true,
+  },
+  misleading: {
+    zeroText: "Accurate",
+    zeroDescription: [],
+    oneText: "Misleading",
+    oneDescription: ["Contains text which is incorrect or misleading"],
+    inverted: true,
+  },
+  helpful: {
+    zeroText: "Unhelful",
+    zeroDescription: [],
+    oneText: "Helpful",
+    oneDescription: ["Completes the task to a high standard"],
+    inverted: false,
+  },
+  creative: {
+    zeroText: "Boring",
+    zeroDescription: [],
+    oneText: "Creative",
+    oneDescription: ["Expresses creativity in responding to the task"],
+    inverted: false,
+  },
+};
+
+export const LabelInputGroup = ({ labelIDs, onChange, isEditable = true }: LabelInputGroupProps) => {
+  const [labelValues, setLabelValues] = useState<number[]>(Array.from({ length: labelIDs.length }).map(() => null));
+
+  const cardColor = useColorModeValue("gray.50", "gray.800");
+
+  return (
+    <Grid templateColumns={"minmax(min-content, 30em)"} rowGap={2}>
+      {labelIDs.map((labelId, idx) => {
+        const { zeroText, oneText, zeroDescription, oneDescription, inverted } = label_messages[labelId];
+
+        let textA = zeroText;
+        let textB = oneText;
+        let descriptionA = zeroDescription;
+        let descriptionB = oneDescription;
+        if (inverted) [textA, textB, descriptionA, descriptionB] = [textB, textA, descriptionB, descriptionA];
```

Comment:
I am not sure why do we need an `inverted` flag, can't we just switch the labels around in `label_messages`?

Reply:
Inverting switches the appearance without changing the values assigned to the different labels. For some labels the backend expects 1 for the negative case and for others it expects 1 for the positive case, the names oneText and zeroText are tied to the meanings that the backend gives. Given we wanted to try having the positive options on the right some of them need to return 1 for the left option, to achieve this the inverted option was added. In the end it seemed clearer to have oneText and zeroText and swap them in one place here rather than have leftText and rightText and need to switch the text manually when we decide that the options should be swapped. Fundamentally it comes down to whether we want to show options in an order not tied to the values and whether the backend will change to put them in the desired order.